Swap PTT state logic to match Hamlib

miek opened the pull request:
Details in #1594

miek committed:
Swap PTT & split-mode state logic to match Hamlib

db4ple commented on mchf-eclipse/drivers/cat/cat_driver.c:
```diff
@@ -1286,10 +1286,10 @@ static void CatDriver_HandleCommands()
+            resp[0]= ts.txrx_mode == TRX_MODE_TX ? tx_state : 0x80;
             if(RadioManagement_IsTxDisabled())
             {
                 resp[0] = 0x00;
```
Please let the code return 0xff here as it was in the original code.

miek commented:
I've re-pushed the commit to return 0xFF in receive as requested, I've also changed it to return 0xFF when TX is disabled as I missed that before.

While testing I noticed that WSJT-X thought I was in split mode, and I found that the logic was swapped there too so I've changed that also. https://github.com/Hamlib/Hamlib/blob/master/yaesu/ft817.c#L567

db4ple commented:
Thanks!  Just saw this myself, when checking the KA7OEI against hamlib: the whole bits on the KA7OEI page are the wrong way it seems. Of course Hamlib is right and our old code was also somewhat right (if not in split mode).
